### ISSUE TITLE
Add failure reason for failed payments

### ISSUE THREAD

**Issue by MrHash** [satbased/satbased#3]

**Comment by johnsBeharry** [comment]
### Merchant Troubleshooting
— credit @pavelex
> In preparation for the next payment [design] sprint, started drafting the troubleshooting flow, when on vs off chain is put side by side it's interesting to see how certain things are easier with LN, however that opens a new infrastructural question that we have to see how and if we include in the payments guide in particular or is it better suited elsewhere.
>
>I have to investigate how different LN implementation handle over-payments, underpayments etc, because LND apparently allows this, which is a bit of a deal breaker for certain use-case - merchants.

![Flowchart(4)](https://user-images.githubusercontent.com/183140/99083229-82e89e00-25c5-11eb-8530-a325d7e3bf58.jpg)

![PaymentsOnChain](https://user-images.githubusercontent.com/183140/99083236-854af800-25c5-11eb-8481-4e19a40c6c56.png)

**Comment by MrHash** [comment]
Interesting flows similar to things i've been through while developing satbased. You might have seen the current event board i produced:

<img width="1003" alt="accounting" src="https://user-images.githubusercontent.com/390925/99084859-63e70d80-25bf-11eb-81b4-f0eb4842f125.png">

The satbased project is designed for "event first" development within the broader concept of DDD. What that means is that the actual entity states are results of the event that occur in "real-life". The diagram describes a `Payment` resource that is specified with two independent state machines depending on direction (incoming/outgoing). This enables us to think of payments in a real world sense simply as "payments in and out". This approach significantly reduces the complexity of the classic deposit/withdrawal concepts and actual aligns better with the original `transaction` concept of bitcoin (i.e. direction agnostic). 

In short the states in the `Payment` entity are reflected by the events, eg. `made`, `reqeuested`, `sent`, `failed`, `cancelled` etc. and seem to be able to encapsulate all the states offered by the various implementations across the bitcoin stack (e.g. bitcoind, various LN impls (incl. hold invoice), elementsd, and even other wallet abstractions). 

Regarding failure reason, the handling of failures depends on a strategy. Satbased supports customizable automated retry/failure strategies, and can also just defer to human decision (hence the need for an explaination phrase).
